jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/compress/PerMessageDeflateExtension.java

```diff
@@ -62,9 +62,12 @@ public void onFrame(Frame frame, Callback callback)
 
         // This extension requires the RSV1 bit set only in the first frame.
         // Subsequent continuation frames don't have RSV1 set, but are compressed.
-        if (OpCode.isDataFrame(frame.getOpCode()) && frame.getOpCode() != OpCode.CONTINUATION)
+        if (OpCode.isDataFrame(frame.getOpCode()))
         {
-            incomingCompressed = frame.isRsv1();
+            if (frame.getOpCode() == OpCode.CONTINUATION)
+                LOG.warn("RSV1 set on CONTINUATION frame");
```

Comment:
Perhaps set a boolean in the extension instance so that we only warn once per instance... which should be once per connection?

Reply:
Can't we just "_Fail the WebSocket Connection_" when this situation arises?

Reply:
Per https://tools.ietf.org/html/rfc7692#page-11

```
   An endpoint MUST NOT set the "Per-Message Compressed" bit of control
   frames and non-first fragments of a data message.  An endpoint
   receiving such a frame MUST _Fail the WebSocket Connection_.
```